rfc-113-expanding-draft-access-for-unauthenticated-users.md

```diff
@@ -0,0 +1,212 @@
+expectations were defined:
+
+- **Live**: This resource is publicly available on live GOV.UK, there are no
+  privacy expectations.
+- **Draft**: This resource is available to users with GOV.UK signon accounts,
+  access is restricted from the wider internet however it is not considered
+  private.
+- **Access limited**: this resource is available to users as the result
+  of implicit permissions granted at the editors discretion (for example,
+  limiting to an organisation in Whitehall) and a limited number of GOV.UK
+  admins. There is an expectation that this resource will be kept private
+  from users who do not meet the criteria.
+
+These were then used to form conclusions on the level of access a
+step-by-step editor should be able to grant to a fact checker.
+
+The editor should be able to grant access to a fact checker to view the
+draft step-by-step navigation page via an auth-bypass token, this should allow
+the fact checker to bypass access limits on the content and any assets created
+for it.
+
+Content that is linked as part of a step-by-step-navigation (and assets that
+are associated with that content) may be published by any publishing application
+and by any organisation. A step-by-step editor can grant a fact checker access
+to live and draft resources, however this cannot be used to provide access to
+access limited content and assets as this would break the expectations of
+privacy.
+
+To provide access across a selection of pages an auth-bypass token would be
+stored in a session cookie this would allow maintaining this token across
+multiple requests without appending a query string (this cookie
+technique is already used for providing an [auth-bypass token to Asset
+Manager][asset-manager-auth-bypass]). To provide the auth-bypass id to
+supporting pages the Publishing API's link expansion system would be used
+for particular document and link types. As there isn't a linking system between
+assets and content it is impossible to maintain accurate links, thus an
+auth-bypass token can grant access to all draft assets for the duration of a
+session. Access limited assets must not be accessible in this circumstance
+unless the asset contains an auth-bypass id that matches the token.
+
+If there is a need to preview access limited content as part of a fact check
+then individual links will have to be provided for that content.
+
+### Consequences
+
+To support this functionality a number of changes are required in the GOV.UK
+stack, these are intended to be non-specific to the step-by-step problem and to
+support similar functionality for future GOV.UK content.
+
+- Authenticating Proxy will be changed to recognise auth-bypass tokens that are
+  present in cookies.
+- Asset Manager will be changed to recognise the presence of a flag in an
+  auth-bypass token that will grant an unauthenticated user access to a draft
+  asset that is not access limited. Fact checkers are not expected to be able
```

Comment:
Is this a flag in the payload part of the JWT token? Just checking my understanding - hadn't heard of JWT before today (NB, have raised https://github.com/alphagov/authenticating-proxy/pull/158 as a result!).

Reply:
Yes it'd be a flag in the token. My spike into it looked like this which probably explains a bit: 

https://github.com/alphagov/collections-publisher/blob/ebf9c7c4d4a9fe1378e9af61cb589c3f3cf6a281/app/lib/jwt_helper.rb#L8-L12

Nice one - opening that PR